### PR TITLE
CARBONDATA-164 Add template for pull requests

### PR DESCRIPTION
**Pull request by ajayyadava**
Added a template for pull requests along the lines of Apache Beam

### CODE REVIEW

**Comment by chenliang613** [comment]
@ajayyadava  looks good of "PR title + tests pass" ,thanks for you started this PR

Suggest adding more information for easily reviewing:
#### What's this PR do?
#### Where should the reviewer start?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+quickly and easily:
+
+ - [ ] Make sure the PR title is formatted like:
+   `[CARBONDATA-<Jira issue #>] Description of pull request`
+ - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
+       Travis-CI on your fork and ensure the whole test matrix passes).
+ - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
+       number, if there is one.
+ - [ ] If this contribution is large, please file an Apache
+       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).
```
Comment (umamaheswararao) [inline .github/PULL_REQUEST_TEMPLATE.md]: Additionally, If contribution is large, how about suggesting to file umbrella JIRA and raise smaller sub JIRAs? Many projects follow this. Always make smaller patches to get reviews easy?

Comment (ajayyadava) [inline .github/PULL_REQUEST_TEMPLATE.md]: Excellent suggestion. Will add it.

**Comment by ajayyadava** [comment]
@chenliang613 

> What does this PR do?
The PR title should give some context and detailed context should be in the JIRA id mentioned in the PR title, right? Also, if users have made sensible and detailed commits, you will get extra information about what the patch does. Makes sense?


> Where should the reviewer start?
I am not sure if I fully understood you, can you please explain what should this contain?

**Comment by chenliang613** [comment]
@ajayyadava ok, for the PR content, suggest adding these sections for easier reviewing:

Why raise this pr?
How to solve(implementation solution)?
How to test?
For example : https://github.com/apache/incubator-carbondata/pull/110

What is your opinion?

**Comment by ajayyadava** [comment]
@chenliang613 

I really like the idea of adding a section on tests added / testing done. 
Why raise this PR? Won't this already be in the JIRA referenced in the PR? 
How to solve? I personally prefer design docs in JIRA but I believe it is a matter of taste and if community feels it is nice to add then we should definitely add it.

I have some time so I will go ahead and address some of the feedback, while we continue to discuss remaining issues.

**Comment by chenliang613** [comment]
LGTM

**Comment by umamaheswararao** [comment]
Hi Ajay, Thanks  for addressing the feedback.
The changes looks good to me.

+1

**Pull request closed without merging**